parsl/dataflow/future_resolution.py

```diff
@@ -55,3 +56,28 @@ def unwrap(v):
 
     type_ = type(iterable)
     return type_(map(unwrap, iterable))
+
+
+@traverse_to_gather.register(Mapping)
```

Comment:
not a fan of the conflation of mapping vs dictionary here... possibly my own personal taste, but this sort of stuff should be as type-specific as possible, mostly because traverse_to_unwrap is making a dictionary, not whatever type T it happened to be called with, and that's some surprise to the user if they're expecting to see a T in their args (as that's what they're passing in...)